### DISCUSSION TITLE
Problem implementing batteryVoltage on custom device (Failed to configure)

### DISCUSSION THREAD

**Discussion by lienbacher** [Koenkk/zigbee2mqtt#11684] [Devices]
Hello!
I am working on a custom sensor and so far most of the things are working great, but i am struggling for a few days already getting batteryVoltage reporting to work properly. 

The device is based on CC2652RB and Z-Stack 5.40. Battery voltage is updated and a read performed on the Attribute in the power config cluster is successful, however binding it in the converter leads to an error:
```
Failed to configure '0x00124b00259c36cb', attempt 2 (Error: ConfigureReporting 0x00124b00259c36cb/8 genPowerCfg([{"attribute":"batteryVoltage","minimumReportInterval":0,"maximumReportInterval":21600,"reportableChange":0}], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'INVALID_VALUE')
    at Endpoint.checkStatus (/Users/username/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:317:28)
    at Endpoint.configureReporting (/Users/username/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:649:22)
    at Object.batteryVoltage (/Users/username/zigbee2mqtt/node_modules/zigbee-herdsman-converters/lib/reporting.js:59:9)
    at Object.configure (/Users/username/zigbee2mqtt/dist/util/externally-loaded.js:20:9)
    at Configure.configure (/Users/username/zigbee2mqtt/lib/extension/configure.ts:115:13))
```

this is the converter:
```const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['Moisture Sensor'], // The model ID from: Device with modelID 'lumi.sens' is not supported.
    model: 'MS1', // Vendor model number, look on the device for a model number
    vendor: '0xD', // Vendor of the device (only used for documentation and startup logging)
    description: 'A simple battery powered sensor', // Description of the device, copy from vendor site. (only used for documentation and startup logging)
    fromZigbee: [fz.temperature, fz.humidity, fz.battery], // We will add this later
    toZigbee: [], // Should be empty, unless device can be controlled (e.g. lights, switches).
    configure: async (device, coordinatorEndpoint, logger) => {
        const firstEndpoint = device.getEndpoint(8);
        await reporting.bind(firstEndpoint, coordinatorEndpoint, ['genPowerCfg', 'msTemperatureMeasurement', 'msRelativeHumidity']);
        const overides = {min: 0, max: 21600, change: 0};
        await reporting.batteryVoltage(firstEndpoint, overides);
        await reporting.batteryPercentageRemaining(firstEndpoint, overides);
        await reporting.temperature(firstEndpoint, overides);
        await reporting.humidity(firstEndpoint, overides);
    },
    exposes: [e.battery(), e.temperature(), e.humidity()], // Defines what this device exposes, used for e.g. Home Assistant discovery and in the frontend
};

module.exports = definition;
```

Every other property reports just fine, if I remove the batteryVoltage binding everything works without an error. I've studied the code and from what I understand I have the impression it should work just fine. 

Any idea what could be wrong? 

(I have also opened [this issue](https://github.com/Koenkk/zigbee-herdsman-converters/issues/3942))

**Comment by lienbacher** [comment]
I managed to launch the debugger via VSCode and I found out that the line that actually fails is line 59 in `herdsman-converters/lib/reporting.js`: `await endpoint.configureReporting('genPowerCfg', p);`, which just so happens to be the line before await `endpoint.read('genPowerCfg', ['batteryVoltage']);`, which I know works because I've done that read call from the dev-console in the frontend. I also know that I call this line in my converter with `await reporting.batteryVoltage(firstEndpoint);`.

Is it possible that configureReporting() of the batteryVoltage attribute fails because according to ZCL specification batteryVoltage is not reportable, and TI-SDK seems to be quite tight to the specifications?

**Comment by Koenkk** [comment]
It could be that the batteryVoltage attribute isn't marked as reportable yet. See here how I enabled this for the ZCL version of the basic cluster: https://github.com/Koenkk/Z-Stack-firmware/blob/6e3b68404136c8cf87979212665e6505bd9a0e3b/router/Z-Stack_3.x.0/firmware.patch#L126

**Reply by lienbacher** [reply]
Thanks!
this was indeed the problem, however just adding the reportable attribute would've been overwritten by the buildprocess of CCStudio, so I edited it in <sdk directory>\source\ti\zstack\ZCL.js and changed the _reportable filed of ATTRID_POWER_CONFIGURATION_BATTERY_VOLTAGE to true and rebuildt the project inside CCS.
